crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs

```diff
@@ -1,162 +1,220 @@
+
+        let updater = self.build_updater(da_p_component, da_d_component);
+
+        self.execute_simulation(
+            capacity,
+            max_block_bytes,
+            fullness_and_bytes,
+            blocks,
+            updater,
         )
-        .1;
-
-    let blocks = l2_blocks.iter().zip(da_blocks.iter()).enumerate();
-
-    let mut updater = AlgorithmUpdaterV1 {
-        min_exec_gas_price: 10,
-        min_da_gas_price: 10,
-        new_exec_price: 10 * gas_price_factor,
-        last_da_gas_price: 5 * gas_price_factor,
-        da_gas_price_factor: NonZeroU64::new(gas_price_factor).unwrap(),
-        l2_block_height: 0,
-        l2_block_fullness_threshold_percent: 50,
-        exec_gas_price_change_percent: 2,
-        max_da_gas_price_change_percent: 10,
-        total_da_rewards: 0,
-        da_recorded_block_height: 0,
-        latest_da_cost_per_byte: 50000,
-        projected_total_da_cost: 0,
-        latest_known_total_da_cost: 0,
-        unrecorded_blocks: vec![],
-        da_p_component,
-        da_d_component,
-        last_profit: 0,
-        second_to_last_profit: 0,
-    };
-
-    let mut gas_prices = vec![];
-    let mut exec_gas_prices = vec![];
-    let mut da_gas_prices = vec![];
-    let mut actual_reward_totals = vec![];
-    let mut projected_cost_totals = vec![];
-    let mut actual_costs = vec![];
-    let mut pessimistic_costs = vec![];
-    for (index, ((fullness, bytes), da_block)) in blocks {
-        let height = index as u32 + 1;
-        exec_gas_prices.push(updater.new_exec_price);
-        let gas_price = updater.algorithm().calculate(max_block_bytes);
-        gas_prices.push(gas_price);
-        // Update DA blocks on the occasion there is one
-
-        if let Some(mut da_blocks) = da_block.clone() {
-            let mut total_costs = updater.latest_known_total_da_cost;
-            for block in &mut da_blocks {
-                block.block_cost *= gas_price_factor;
-                total_costs += block.block_cost;
-                actual_costs.push(total_costs);
-            }
-            updater.update_da_record_data(da_blocks.to_owned()).unwrap();
-            assert_eq!(total_costs, updater.projected_total_da_cost);
-            assert_eq!(total_costs, updater.latest_known_total_da_cost);
-        }
+    }
+
+    fn build_updater(
+        &self,
+        da_p_component: i64,
+        da_d_component: i64,
+    ) -> AlgorithmUpdaterV1 {
+        let gas_price_factor = 100;
+        let updater = AlgorithmUpdaterV1 {
+            min_exec_gas_price: 10,
+            min_da_gas_price: 10,
+            new_scaled_exec_price: 10 * gas_price_factor,
+            // last_da_gas_price: *starting_da_gas_price,
+            last_da_gas_price: 100,
+            gas_price_factor: NonZeroU64::new(gas_price_factor).unwrap(),
+            l2_block_height: 0,
+            l2_block_fullness_threshold_percent: 50,
+            exec_gas_price_change_percent: 2,
```

Comment:
It would be nice to be able to read more of those parameters from cmd line. Could be in a different PR, though.